src/components/AutoCompleteInput.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverBody,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { Box, Text } from "@chakra-ui/react";
+import React, { useCallback, useEffect } from "react";
+import { useRef, useState } from "react";
+
+type AutoCompleteProps = {
+  children: React.ReactNode;
+  triggerRef: React.RefObject<HTMLTextAreaElement>;
+  parentRef: React.RefObject<HTMLDivElement>;
+  availablePrompts: {
+    command: string;
+    helpTitle: string;
+    helpDescription: string;
+  }[];
+};
+
+const AutoCompleteInput = ({
+  children,
+  triggerRef,
+  parentRef,
+  availablePrompts,
+}: AutoCompleteProps) => {
+  const [popupPosition, setPopupPosition] = useState<number>(0);
+  const [inputWidth, setInputWidth] = useState<number>(0);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [selectedIndex, setSelectedIndex] = useState<number>(-1);
+  const suggestionRefs = useRef<(HTMLDivElement | null)[]>([]);
+  // const triggerRef = useRef<HTMLInputElement>(null);
+
+  const bgColor = useColorModeValue("white", "gray.700");
+  const hoverBg = useColorModeValue("gray.200", "gray.600");
+
+  const [suggestions, setSuggestions] = useState<
+    { command: string; helpTitle: string; helpDescription: string }[]
+  >([]);
+
+  // Set possition and width
+  useEffect(() => {
+    if (triggerRef.current && parentRef.current) {
+      const parentRect = parentRef.current.getBoundingClientRect();
+      setInputWidth(parentRect.width);
+      setPopupPosition((parentRect.left + window.scrollX) / 10);
+    }
+  }, [parentRef, suggestions.length, triggerRef]);
```

Comment:
```suggestion
  }, [parentRef, triggerRef]);
```

`suggestions` is not a dependency of this `useEffect`